spec/models/prison_spec.rb

```diff
@@ -84,13 +84,25 @@
     describe '.nomis_ids' do
       before do
         allow(described_class).to receive(:all).
-          and_return([double('prison', nomis_id: 'WSI').as_null_object,
-                      double('prison', nomis_id: nil).as_null_object])
+          and_return([object_double(described_class.new, nomis_id: 'WSI').as_null_object,
+                      object_double(described_class.new, nomis_id: nil).as_null_object])
       end
       it 'returns nomis ids for all prisons with a nomis id' do
         expect(described_class.nomis_ids).to match_array(['WSI'])
       end
     end
+
+    describe '.enabled_nomis_ids' do
+      before do
+        allow(described_class).to receive(:all).
```

Comment:
I think it might be better to stub the Rails config here as opposed to this specific class method -`.all`. We are revealing knowledge of the implementation here where as I feel stubbing the `data store` doesn't do this as much as its the source of truth for the models. We are saying that we know that `enabled` must call `all` before doing its filtering.